### DISCUSSION TITLE
Optionally prefix computed column procedures with the table name

### DISCUSSION THREAD

**Discussion by TimoStolz** [PostgREST/postgrest#2507] [Q&A] [answered]
### Environment

* PostgreSQL version: 14.5
* PostgREST version: 10.0
* Operating system: Ubuntu 22.04

### Description of issue

Quite often, I use Postgrest in combination with [Graphile](https://www.graphile.org/), because most of my customers need both, a REST API and a Graphql API. Unfortunately, Postgrest and Graphile are following different naming conventions for computed columns or computed relationships (please see here for [Postgrest](https://postgrest.org/en/stable/api.html#computed-virtual-columns) and [Graphile](https://www.graphile.org/postgraphile/computed-columns/)).

```sql
CREATE TABLE people (
  fname text,
  lname text
);

-- Postgrest convention, without any prefix.
CREATE FUNCTION full_name(people) RETURNS text AS $$
  SELECT $1.fname || ' ' || $1.lname;
$$ LANGUAGE SQL;

-- Graphile convention, prefixed by the table name.
CREATE FUNCTION people_full_name(people) RETURNS text AS $$
  SELECT $1.fname || ' ' || $1.lname;
$$ LANGUAGE SQL;
```

Except from the prefix, almost all functions could be compatible. (There are other differing conventions, but it's quite easy to satisfy them at the same time in most cases.) That's why I would love to see support for an optional table-name-prefix for computed column procedures in Postgrest. (This prefix should be skipped from the API.)

**Comment by wolfgangwalther** [comment, marked as answer]
Hm. That prefix would be quite contrary to our naming conventions.

Maybe you can solve it like this:
- Create the `full_name(...)` function only
- Create an event trigger that reacts on the create statement. When a function is created that matches a few criteria (in the api schema, single table-type argument), you issue another create function statement, similar to this:

```sql
CREATE FUNCTION <table_name>_<func_name>(<table_name>) RETURNS <return_type> AS $$
  SELECT <func_name>($1);
$$;
```

This way you only need to maintain one set of functions, but have both?
